components/collections-panel/collections-panel-component.js

```diff
@@ -0,0 +1,147 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { toastr } from 'react-redux-toastr';
+
+import CollectionPanelItem from './collections-panel-item/collections-panel-item-component';
+
+// constants
+import { FAVOURITES_COLLECTION } from './collections-panel-constants';
+
+// styles
+import styles from './collections-panel-styles.scss';
+
+class CollectionsPanel extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      newCollection: null // new collection's name
+    };
+  }
+
+  onAddCollection() {
+    const { addCollection } = this.props;
+    const { newCollection } = this.state;
+
+    if ((newCollection || '').toLowerCase() === 'favourites') {
+      toastr.error('Duplicated Favourites list', 'You cannot duplicate this list.');
+      return;
+    }
+
+    addCollection(newCollection);
+  }
+
+  onToggleFavourite() {
+    const { toggleFavourite, favourites, resource, resourceType } = this.props;
+    const favourite = favourites.find(fav => fav.resourceId === resource.id) || {};
+    toggleFavourite(favourite, { resourceId: resource.id, resourceType });
+  }
+
+  onToggleResource(isAdded, collection) {
```

Comment:
With `onToggleResource` do you mean `onToggleCollection`? If it is that case, please change the name. Let's try to not change this names a be more descriptive

Reply:
Yep, my bad here.